@xen-orchestra/lite/scripts/release.mjs

```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env zx
+
+import argv from "minimist";
+import { tmpdir } from "os";
+
+$.verbose = false;
+
+const DEPLOY_SERVER = "www-xo.gpn.vates.fr";
+
+const { version: pkgVersion } = await fs.readJson("./package.json");
+
+const opts = argv(process.argv, {
+  boolean: ["help", "build", "deploy", "ghRelease", "tarball"],
+  string: [
+    "base",
+    "dist",
+    "ghToken",
+    "tarballDest",
+    "tarballName",
+    "username",
+    "version",
+  ],
+  alias: {
+    u: "username",
+    h: "help",
+    "gh-release": "ghRelease",
+    "gh-token": "ghToken",
+    "tarball-dest": "tarballDest",
+    "tarball-name": "tarballName",
+  },
+  default: {
+    dist: "dist",
+    version: pkgVersion,
+  },
+});
+
+let {
+  base,
+  build,
+  deploy,
+  dist,
+  ghRelease,
+  ghToken,
+  help,
+  tarball,
+  tarballDest,
+  tarballName,
+  username,
+  version,
+} = opts;
+
+const usage = () => {
+  console.log(
+    `Usage: ./release.mjs
+    [--help|-h - show this message]
+
+    [--version X.Y.Z - XO Lite version - default: package.json version]
+    [--dist /path/to/folder - build destination folder - default: dist]
+
+    [
+      --build - whether to build XO Lite or not
+      [--base url - base URL for assets - default: "/" or "lite.xen-orchestra.com/dist" if --deploy is passed]
+    ]
+
+    [
+      --tarball - whether to generate a tarball or not
+      [--tarball-dest /path/to/folder - tarball destination folder]
+      [--tarball-name file.tar.gz - tarball file name - default xo-lite-X.Y.Z.tar.gz]
+    ]
+
+    [
+      --gh-release - whether to release on GitHub or not
+      [--gh-token token - GitHub API token with "Contents" write permissions]
+    ]
+
+    [
+      --deploy - whether to deploy to xen-orchestra.com or not
+      --username|-u <LDAP username>
+    ]
+`
+  );
+};
+
+if (help) {
+  usage();
+  process.exit(0);
+}
+
+const yes = async (q) =>
+  ["y", "yes"].includes((await question(q + " [y/N] ")).toLowerCase());
+
+const no = async (q) => !(await yes(q));
```

Comment:
Yes, if you type `yas`, it will be handled as "no".

I think it's safer that way and I also think it's a common behavior because the question almost always looks like "Do you want to keep doing this even though it will have those side effects?" so "no" is the safe-abort option and it's ok if we choose "no" by mistake, but it's not ok if we choose "yes" by mistake.

e.g.: if you typo "noo" in this case, it will have unwanted consequences:
```js
if (await no('Do you want to continue?') {
  safelyAbort()
} else {
  proceedWithSideEffects()
}
```

Reply:
Is it hard to prompt the user again for a response if it's neither `y/yes` nor `n/no`? Similar to the behavior GitHub uses for `git add -p` for example.